test-reflection-cpp.cpp

```diff
@@ -68,6 +68,61 @@ TEST_CASE("nested", "[reflection]")
     CHECK(result == R"(a=1 b=2 c=3 d="hello" e={name="John Doe" email="john@doe.com" age=42})");
 }
 
+TEST_CASE("EnumerateMembers.ind_and_value", "[reflection]")
+{
+    auto ps = Person { "John Doe", "john@doe.com", 42 };
+    Reflection::EnumerateMembers(ps, []<size_t I>(auto&& value) {
+        if constexpr (I == 0)
+        {
+            CHECK(value == "John Doe");
+        }
+        else if constexpr (I == 1)
+        {
+            CHECK(value == "john@doe.com");
+        }
+        else if constexpr (I == 2)
+        {
+            CHECK(value == 42);
+        }
+    });
+
+}
+
+
+TEST_CASE("EnumerateMembers.ind_and_type", "[reflection]")
+{
+    Reflection::EnumerateMembers<Person>([]<auto I, typename T>() {
+            if constexpr(I==0)
+            {
+                static_assert(T{} == "");
+            }
+            if constexpr(I==1)
+            {
+                static_assert(T{} == "");
+            }
+            if constexpr(I==2)
+            {
+                static_assert(T{} == 0);
+            }
+    });
+
```

Comment:
```suggestion
    Reflection::EnumerateMembers<Person>([]<auto I, typename T>() {
            if constexpr(I==0)
            {
                static_assert(std::same_as<T, std::string_view>);
            }
            if constexpr(I==1)
            {
                static_assert(std::same_as<T, std::string>);
            }
            if constexpr(I==2)
            {
                static_assert(std::same_as<T, int>);
            }
    });
```

These static asserts should be more expressive & accurate, and the empty line at the end was removed as well :)